# 事件

## error 事件

+ Worker 接口的 error 事件会在 worker 发生错误时触发

  ```js
  const myWorker = new Worker("worker.js");

  myWorker.onerror = (event) => {
    console.log("There is an error with your worker!");
  };
  ```

## message 事件

+ 当 worker 的父级接收到来自其 worker 的消息时（也就是说，当 worker 通过 DedicatedWorkerGlobalScope.postMessage() 发送消息时），会在 Worker 对象上触发 message 事件。

+ 此事件不能取消，也不会冒泡

  ```js
  const worker = new Worker("static/scripts/worker.js");

  worker.addEventListener("message", (event) => {
    console.log(`Received message from worker: ${event.data}`);
  });

  // worker 使用 self.postMessage() 发出消息
  // static/scripts/worker.js

  self.postMessage("I'm alive!");
  ```

## messageerror 事件

+ 当 Worker 对象接收到一条无法被反序列化的消息时，将在该对象上触发 messageerror 事件

+ 此事件不能取消，也不会冒泡

  ```js
  const worker = new Worker("static/scripts/worker.js");

  worker.addEventListener("message", (event) => {
    console.error(`Received message from worker: ${event}`);
  });

  worker.addEventListener("messageerror", (event) => {
    console.error(`Error receiving message from worker: ${event}`);
  })
  ```
